lib/liquid/context.rb

```diff
@@ -27,7 +27,7 @@ def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_erro
       @this_stack_used = false
 
       if rethrow_errors
-        self.exception_handler = ->(e) { raise }
+        self.exception_handler = ->(e) { raise e }
```

Comment:
Imagine `x | undefined | escape`. However I think I agree that the filter chain should be interrupted **and** the result forced to `nil`.